### PR TITLE
[DAGS vizualizer] send initial data on new ws connection

### LINKED ISSUE #1911: Visualizer: Keep recent history of message/transaction/branch
Keep recent history and send it to the new ws client, same as the visualizer in dashboard.

### PR DESCRIPTION
**Pull request by acha-bill**
# Description of change

Fixes #1911 

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [ ] My code follows the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: plugins/dagsvisualizer/visualizer.go [hunk 1]
```diff
@@ -22,6 +23,9 @@ var (
 	visualizerWorkerCount     = 1
 	visualizerWorkerQueueSize = 500
 	visualizerWorkerPool      *workerpool.NonBlockingQueuedWorkerPool
+	maxEventsBufferSize       = 200
```
Comment (jkrvivian) [inline plugins/dagsvisualizer/visualizer.go:26]: Rename it to `maxWsMessageBufferSize`

File: plugins/dagsvisualizer/visualizer.go [hunk 2]
```diff
@@ -349,3 +371,12 @@ func newBranchVertex(branchID ledgerstate.BranchID) (ret *branchVertex) {
 	})
 	return
 }
+
+func storeMessage(msg *wsMessage) {
```
Comment (jkrvivian) [inline plugins/dagsvisualizer/visualizer.go:375]: Rename it to `storeWsMessage`

File: plugins/dagsvisualizer/ws.go [hunk 3]
```diff
@@ -116,3 +118,23 @@ func websocketRoute(c echo.Context) error {
...
+			log.Errorf("failed to send DAG message to client: %s", err.Error())
+		}
+	}
+}
+
+func sendJSON(ws *websocket.Conn, msg *wsMessage) error {
+	if err := ws.WriteJSON(msg); err != nil {
+		return err
+	}
+	if err := ws.SetWriteDeadline(time.Now().Add(webSocketWriteTimeout)); err != nil {
```
Comment (jkrvivian) [inline plugins/dagsvisualizer/ws.go:136]: The deadline setting should go before `WriteJSON`

**Comment by jkrvivian** [comment]
@acha-bill Since we're replaying all types of WSMessage, could you make sure that we check the existence of a msg/tx/branch in the front-end handler? especially when establishing parents' edge.

**Comment by acha-bill** [comment]
> @acha-bill Since we're replaying all types of WSMessage, could you make sure that we check the existence of a msg/tx/branch in the front-end handler? especially when establishing parents' edge.

I think this has already been covered in the existing code.

**Review by jkrvivian — APPROVED** [review]

**Pull request merged**